_source/_posts/2019-12-26-java-mongodb-crud.md

```diff
@@ -0,0 +1,519 @@
+```
+
+Check out the inventory now and you should see these new items.
+
+```bash
+$ http :8080/fridge
+
+HTTP/1.1 200
+Content-Type: application/hal+json;charset=UTF-8
+Date: Mon, 16 Dec 2019 03:45:23 GMT
+Transfer-Encoding: chunked
+
+{
+    "_embedded": {
+        "fridge": [
+            {
+                "_links": {
+                    "refrigeratorItem": {
+                        "href": "http://localhost:8080/fridge/5dae7b4c6a99f01364de916c"
+                    },
+                    "self": {
+                        "href": "http://localhost:8080/fridge/5dae7b4c6a99f01364de916c"
+                    }
+                },
+                "expiration": "01-01-2020",
+                "name": "milk",
+                "owner": "Andrew"
+            },
+            {
+                "_links": {
+                    "refrigeratorItem": {
+                        "href": "http://localhost:8080/fridge/5dae7b4d6a99f01364de916d"
+                    },
+                    "self": {
+                        "href": "http://localhost:8080/fridge/5dae7b4d6a99f01364de916d"
+                    }
+                },
+                "expiration": "02-10-2020",
+                "name": "cheese",
+                "owner": "Andrew"
+            },
+            {
+                "_links": {
+                    "refrigeratorItem": {
+                        "href": "http://localhost:8080/fridge/5dae7b4f6a99f01364de916e"
+                    },
+                    "self": {
+                        "href": "http://localhost:8080/fridge/5dae7b4f6a99f01364de916e"
+                    }
+                },
+                "expiration": "03-30-2020",
+                "name": "pizza",
+                "owner": "Andrew"
+            }
+        ]
+    },
+    "_links": {
+        "profile": {
+            "href": "http://localhost:8080/profile/fridge"
+        },
+        "self": {
+            "href": "http://localhost:8080/fridge{?page,size,sort}",
+            "templated": true
+        }
+    },
+    "page": {
+        "number": 0,
+        "size": 20,
+        "totalElements": 3,
+        "totalPages": 1
+    }
+}
+
+```
+
+Notice that the returns JSON gives you the URL for each individual item. If you wanted to delete the first item in the list above, you could run the following request.
+
+```bash
+http DELETE :8080/fridge/5dae7b4c6a99f01364de916c
+```
+
+The long string,`5dae7b4c6a99f01364de916c`, is the unique ID for that item. MongoDB doesn't use sequential integer ID numbers like SQL databases often do. It uses randomly generated UUIDs instead.
+
+If you wanted to update an item, you could use a PUT, as shown below.
+
+```bash
+http PUT :8080/fridge/5dae7b4f6a99f01364de916e name="old pizza" expiration="03-30-2020" owner="Erin"
+```
+
+Note that with a PUT you have to send data for all the fields, not just the field you want to update, otherwise the omitted fields are set to null. If you just want to update select fields, use a PATCH.
```

Comment:
```suggestion
Note that with a PUT you have to send data for all the fields, not just the field you want to update, otherwise the omitted fields are set to `null`. If you just want to update select fields, use a PATCH.
```